rewrite-java/src/main/java/org/openrewrite/java/ChangePackage.java

```diff
@@ -21,22 +21,44 @@
 import org.openrewrite.Option;
 import org.openrewrite.Recipe;
 import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
 import org.openrewrite.internal.lang.Nullable;
-import org.openrewrite.java.tree.J;
-import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
 
 import java.nio.file.Paths;
 
+import static org.openrewrite.Tree.randomId;
+
+/**
+ * A recipe that will rename a package name in package statements, imports, and fully-qualified types (see: NOTE).
+ *
+ * NOTE: Does not currently transform all possible type references, and accomplishing this would be non-trivial.
+ * For example, a method invocation select might refer to field `A a` whose type has now changed to `A2`, and so the type
+ * on the select should change as well. But how do we identify the set of all method selects which refer to `a`? Suppose
+ * it were prefixed like `this.a`, or `MyClass.this.a`, or indirectly via a separate method call like `getA()` where `getA()`
+ * is defined on the super class.
+ */
 @Getter
 @RequiredArgsConstructor
 @AllArgsConstructor(onConstructor_ = @JsonCreator)
 @EqualsAndHashCode(callSuper = true)
 public class ChangePackage extends Recipe {
```

Comment:
We have a "renameVariable" and I believe this should be "renamePackage" to better articulate what this recipe is doing. It is consistent with how the IDEs represent this change as well.